### PR TITLE
Make GetMethods() return methods in declaration order consistantly

### PR DESCRIPTION
**Pull request by vancem**
For classes with less than 64 methods, the GetMethods() call returns the methods in the order
of delcaration.   However internally in the runtime we read methods in chunks and put
them on a linked list, which means that for classes with more than this number of methods
the order gets scrambled.

There are classes like System.Diagnostic.Tracing.EventSource that depend on reflection
returning the methods in order of declaration IN CERTAIN SCENARIOS, and this does work
today for classes with less than 64 methods.     However we are now seeing cases where people
are relying on this scenario with classes of more than 64 methods.

It is pretty trivial to fix this so that we preserve the order of the methods.   Basicaly we
reverse the list of chunks after we have generated it, which restores the order.

This fix can be viewed as a compatibility fix, but in general preserving delclaration order
is a useful, because it is a natural way of resolving conflicts (e.g. last one wins), and
there is no other way to get this information from reflection unless we expose it in
GetMethods().     we are 95% of the way there (it works in chunks), this simple
change pushes us to 100% (for statically declared methods).

### CODE REVIEW

File: src/vm/class.cpp [hunk 1]
```diff
@@ -3114,6 +3114,24 @@ void EEClass::AddChunk (MethodDescChunk* pNewChunk)
 }
 
 //*******************************************************************************
+// Reverses the linked list of MethodDescChunks.  Used to keep the methods
+// in the same order as the meta-data so that reflection also returns
+// methods in lexical order.  
+void EEClass::ReverseChunks()
+{
+	PTR_MethodDescChunk chunks = GetChunks();
```
Comment (janvorli) [inline src/vm/class.cpp]: A nit - you seem to have tabs insted of spaces in all of the new code.

**Comment by janvorli** [comment]
LGTM modulo the nit

**Comment by vancem** [comment]
Sorry, my editor was not configured to remove tabs.   Fixed.

**Comment by jkotas** [comment]
This does not work well with fragile NGen - fragile NGen reorders the methods (for hot/cold splitting, etc.).

**Comment by vancem** [comment]
Offline Jan suggested that we make GetMethods go directly back to meta-data to avoid any issues with ordering for compilation cases.     Thus I am going to withdraw this pull request (since it would work completely differently than this code delta).   I have logged this bug to track.  

Bug 228031*:Make GetMethods() return methods in declaration order consistantly

**Pull request closed without merging**